Fast and Accurate GPU Quantization for Transformers.md

```diff
@@ -38,53 +23,52 @@ We describe an approach to GPU-based quantization of transformers that is both *
 	* [Fusion Implementation](#Fusion%20Implementation)
 * [References](#References)
 
+# Part I: Accurate Quantization
 
-## Background
-
-We’ll begin with a super quick summary of quantization. For further reading on this subject, we’ve listed some nice blogs/papers in our [References](#References) section [1-4].
+We’ll begin with a quick overview of quantization. For deeper reading on this subject, some nice blogs/papers in the [References](#References) section [1-4].
 
 ### The Quantization Equation
-In principle, we can use any function to convert from a higher-precision to lower-precision representation. But a linear function is simplest and quickest on the hardware [4]:
 
-$$Q(r)=\textrm{Int}(r/S)-Z$$
+In principle, we can use any function to convert from a higher-precision to lower-precision representation. But a linear function is simplest and quickest [4]:
 
-Here, $Q, r$ are the INT output and FP input, while $S, Z$ are a scale factor and bias. $\textrm{Int}$ is a function that rounds to the nearest integer. To minimize inference cost, we can set $Z=0$ (we do this for our GPU implementation). 
+$$Q(x)=\textrm{Int}(x/S)-Z$$ 
 
-The corresponding dequantization equation is even simpler:
+Here, $Q$ and $x$ are the fixed-point output and floating-point input, while $S$ and $Z$ represent the scale factor and bias. $\textrm{Int}$ is a function that rounds to the nearest integer, clipping values outside of the representable range. 
 
-$$\tilde{r}=S Q(r)$$
+After applying our lower-precision operation we return the data to it's original dynamic range by dequantizing the data: 
+
+$$\tilde{x}=S Q(x) + Z$$
 
 This method is called **uniform quantization** since the quantized values are uniformly distributed over the input space. To calculate $S$ we select a **clipping range** $[\alpha, \beta]$ and then use:
 
 $$S=\frac{\beta-\alpha}{2^b-1}$$
 
-Here, $b$ is the number of bits in our quantization scheme. We choose to enforce $\alpha=-\beta$, which is known as **symmetric quantization**. This simplifies the (de)quantization function by setting $Z=0$.
+Here, $b$ is the number of bits in our quantization scheme. GPU based quantization schemes typicall enforce $\alpha=-\beta$, which is known as **symmetric quantization**. This simplifies the (de)quantization functions by setting $Z=0$, which helps reduce the cost of the transformation {do we have any reference for this??}.
```

Comment:
```suggestion
Here, $b$ is the number of bits in our quantization scheme. GPU based quantization schemes typically enforce $\alpha=-\beta$, which is known as **symmetric quantization**. This simplifies the (de)quantization functions by setting $Z=0$, which helps reduce the cost of the transformation {do we have any reference for this??}.
```

Reply:
Yes, we can use one of the references we already have: https://arxiv.org/pdf/2103.13630.pdf